src/debugAdapter/goDebug.ts

```diff
@@ -995,49 +995,67 @@ class GoDebugSession extends DebugSession {
 	protected variablesRequest(response: DebugProtocol.VariablesResponse, args: DebugProtocol.VariablesArguments): void {
 		verbose('VariablesRequest');
 		let vari = this._variableHandles.get(args.variablesReference);
-		let variables;
+		let variablesPromise: Promise<DebugProtocol.Variable[]>;
+		let loadChildren = async (exp: string, v: DebugVariable) => {
+			if (v.len > 0 && v.children.length === 0) {
+				// from https://github.com/derekparker/delve/blob/master/Documentation/api/ClientHowto.md#loading-more-of-a-variable
+				await this.evaluateRequestImpl({ 'expression': exp }).then(result => {
```

Comment:
since we are inside an async function I think we dont need then, i.e. we can do
```ts
await this.X()
const variable = this.delve.isApiV1 ? <DebugVariable>result : (<EvalOut>result).Variable;
v.children = variable.children;
```

Reply:
In your example, what happens if `this.X()` fails, what will the value of `result` be?

Reply:
depends what you mean by fail. if it throws an error it will throw an error but that will be the same for then is you passed only the onFulfilled function

Reply:
well if by fail you mean throw an error it will be the same thing(the default onRejection just throws up the error). Also FYI its deprecated as of now to have [unhandled promise rejections and in the future it will cause the Node process to crash](https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections)

so in this case you can either wrap the await in a try catch block and do nothing in the catch or provide a onrejected method that does nothing

Reply:
Oh good point will add an `onrejected` method, is `reason => reason` sufficient to do nothing?